### PR TITLE
[Kube-Gen] Generate Grafana and Prom Scripts in Kubernetes

### PR DESCRIPTION
**Pull request by Lewiky**
Generates the deployments and services needed to run kube and prom in kubernetes.

Lots of tests changes - sorry!

### CODE REVIEW

File: src/main/scala/temple/builder/OrchestrationBuilder.scala [hunk 1]
```diff
@@ -23,7 +23,10 @@ object OrchestrationBuilder {
             name = kebabName,
             image = dockerImage,
             dbImage = dbImage.toString,
-            ports = Seq(("api", port.service)),
+            ports = Seq(
```
Comment (jaylees14) [inline src/main/scala/temple/builder/OrchestrationBuilder.scala]: What about just storing the type `Ports` here?

File: src/main/scala/temple/generate/orchestration/KongConfigGenerator.scala [hunk 2]
```diff
@@ -13,7 +13,7 @@ object KongConfigGenerator {
 
   /** Generate a kong service declaration for a Temple service */
   private def generateServiceDef(service: Service): String = {
-    val urls = service.ports map {
+    val urls = service.ports.headOption map {
```
Comment (jaylees14) [inline src/main/scala/temple/generate/orchestration/KongConfigGenerator.scala]: If you store a `Ports` type above, you don't have to do this unsafe lookup that depends on ordering

**Review by jaylees14 — APPROVED** [review]
Nice one, just one suggestion but looks good.

File: src/test/scala/temple/generate/orchestration/kube/KubernetesGeneratorTest.scala [hunk 3]
```diff
@@ -104,4 +104,17 @@ class KubernetesGeneratorTest extends FlatSpec with Matchers {
...
+    val output = KubernetesGenerator.generate("example", UnitTestData.basicOrchestrationRootWithMetrics)
+    output.keys should contain(File("kube/prom", "prometheus-deployment.yaml"))
+    output.keys should contain(File("kube/prom", "prometheus-service.yaml"))
+    output.keys should contain(File("kube/grafana", "grafana-deployment.yaml"))
+    output.keys should contain(File("kube/grafana", "grafana-service.yaml"))
+
+    output(File("kube/prom", "prometheus-deployment.yaml")) shouldBe UnitTestData.userPromDeployment
+    output(File("kube/prom", "prometheus-service.yaml")) shouldBe UnitTestData.userPromService
+    output(File("kube/grafana", "grafana-deployment.yaml")) shouldBe UnitTestData.userGrafanaDeployment
+    output(File("kube/grafana", "grafana-service.yaml")) shouldBe UnitTestData.userGrafanaService
```
Comment (c-harding) [inline src/test/scala/temple/generate/orchestration/kube/KubernetesGeneratorTest.scala:118]: what’s the point of the first four lines?

**Review by c-harding — APPROVED** [review]

**Pull request merged**